modules/orchestrator/src/main/java/org/eclipse/xpanse/orchestrator/utils/CredentialApiUtil.java

```diff
@@ -0,0 +1,309 @@
+            if (!Objects.isNull(availableCredentialTypes) && !Objects.isNull(
+                    credentialDefinitions)) {
+                Map<CredentialType, CredentialDefinition> credentialMap =
+                        credentialDefinitions.stream()
+                                .filter(credentialDefinition -> availableCredentialTypes.contains(
+                                        credentialDefinition.getType()))
+                                .collect(Collectors.toMap(AbstractCredentialInfo::getType,
+                                        cd -> (CredentialDefinition) cd));
+                deleteCredentialApi(csp);
+                generateCredentialApi(credentialMap, csp);
+            }
+        });
+    }
+
+    /**
+     * Get serviceUrl.
+     *
+     * @return serviceUrl
+     */
+    public String getServiceUrl() {
+        return openApiUtil.getServiceUrl();
+    }
+
+    /**
+     * Get credentialApi Url.
+     *
+     * @return credentialApi Url
+     */
+    public String getCredentialApiUrl(Csp csp) {
+        if (openapiPath.endsWith("/")) {
+            return getServiceUrl() + "/" + openapiPath + csp + "_" + "credentialApi" + ".html";
+        }
+        return getServiceUrl() + "/" + openapiPath + "/" + csp + "_" + "credentialApi" + ".html";
+    }
+
+    /**
+     * Get the work directory of the credentialApi.
+     *
+     * @return workdir  The work directory of the credentialApi.
+     */
+    private String getCredentialApiDir() {
+        return openApiUtil.getOpenApiWorkdir();
+    }
+
+    /**
+     * create credentialApi for plugins.
+     */
+    public String createCredentialApi(Map<CredentialType, CredentialDefinition> credentialMap,
+                                      Csp csp) {
+        String yamlFileName = csp + "_" + "credentialApi" + ".yaml";
+        String credentialApiDir = getCredentialApiDir();
+        File dir = new File(credentialApiDir);
+        if (!dir.exists()) {
+            log.info("Create service openApi dir:{} success:{}", credentialApiDir, dir.mkdirs());
+        }
+        File yamlFile = new File(credentialApiDir, yamlFileName);
+        File htmlFile = new File(credentialApiDir, csp + "_" + "credentialApi" + ".html");
+        try {
+            String apiDocsJson = getApiDocsJson(credentialMap, csp);
+            try (FileWriter apiWriter = new FileWriter(yamlFile.getPath())) {
+                apiWriter.write(apiDocsJson);
+            }
+            log.info("{} credentialApi yamlFile:{} create success.", csp, yamlFile.getPath());
+            if (yamlFile.exists() && openApiUtil.downloadClientJar(credentialApiDir)) {
+                File jarPath = new File(credentialApiDir, "openapi-generator-cli.jar");
+                String comm = String.format("java -jar %s generate -g html2 "
+                        + "-i %s -o %s", jarPath.getPath(), yamlFile.getPath(), credentialApiDir);
+                Process exec = Runtime.getRuntime().exec(comm);
```

Comment:
```suggestion
                Process exec = Runtime.getRuntime().exec(comm);
```
please avoid using deprecated method.